modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-options/src/main/resources/META-INF/resources/options_field.js

```diff
@@ -338,6 +344,28 @@ AUI.add(
 						);
 					},
 
+					_afterEditingLanguageIdChange: function() {
+						var instance = this;
+
+						var defaultLanguage = instance._getDefaultLanguageId();
+						var editingLanguage = instance._getCurrentEditingLanguageId();
+
+						var value = instance.get('value');
+
+						if (editingLanguage != defaultLanguage) {
+							if (value) {
+								if (!value[editingLanguage]) {
```

Comment:
Invert this if condition, because is more readable, when you have a if / else condition, you could put just `value[editingLanguage]`